src/layout/PaymentDetails/PaymentDetailsComponent.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { useEffect } from 'react';
+
+import { useQueryClient } from '@tanstack/react-query';
+
+import { FD } from 'src/features/formData/FormDataWrite';
+import { useInstanceIdParams } from 'src/hooks/useInstanceIdParams';
+import { useOrderDetailsQuery } from 'src/layout/Payment/queries/useOrderDetailsQuery';
+import { PaymentDetailsTable } from 'src/layout/PaymentDetails/PaymentDetailsTable';
+import type { PropsFromGenericComponent } from 'src/layout';
+
+export type IPaymentDetailsProps = PropsFromGenericComponent<'PaymentDetails'>;
+
+export function PaymentDetailsComponent({ node }: IPaymentDetailsProps) {
+  const { partyId, instanceGuid } = useInstanceIdParams();
+  const { title, description } = node.item.textResourceBindings || {};
+  const hasUnsavedChanges = FD.useHasUnsavedChanges();
+  const queryClient = useQueryClient();
+  const { data: orderDetails } = useOrderDetailsQuery(partyId, instanceGuid);
+
+  useEffect(() => {
+    if (!hasUnsavedChanges) {
+      queryClient.invalidateQueries({ queryKey: ['fetchOrderDetails'] });
+    }
```

Comment:
Seems like the backend should notify us when order details change as a result of form data changes instead. Invaliding the request and fetching it again after each save operation seems excessive.

At some point, if we want to provide an expression function for getting payment details, we'd have to move this effect to work on all form pages in every app, and then we'd just double the amount of requests we send back and forth on form saving.

Reply:
How would backend notify us though? Theres no websockets/SSE stuff right?

Reply:
Nah, but when we're saving form data (i.e. a data model, with the PATCH endpoint) we already get a list of current validation messages from the backend. This, of course, has the effect of collecting pretty much all functionality into the same endpoint - but I think it's better that the backend at least hints that cache should be invalidated than just always invalidating the cache every time we save.